src/devices/Nmea0183/TalkerSentence.cs

```diff
@@ -0,0 +1,217 @@
+            }
+
+            TalkerId talkerId = new TalkerId(sentence[1], sentence[2]);
+            SentenceId sentenceId = new SentenceId(sentence[3], sentence[4], sentence[5]);
+
+            string[] fields = sentence.Substring(SentenceHeaderLength).Split(',');
+            int lastFieldIdx = fields.Length - 1;
+            (byte? checksum, string lastField) = GetChecksumAndLastField(fields[lastFieldIdx]);
+            fields[lastFieldIdx] = lastField;
+            
+            TalkerSentence result = new TalkerSentence(talkerId, sentenceId, fields);
+
+            if (checksum.HasValue)
+            {
+                byte realChecksum = CalculateChecksumFromSentenceString(sentence);
+
+                if (realChecksum != checksum.Value)
+                {
+                    throw new InvalidOperationException($"Checksum in the sentence (0x{checksum:X2}) does not match calculated checksum (0x{realChecksum:X2}) for sentence `{sentence}`");
+                }
+            }
+
+            return result;
+        }
+
+        /// <summary>
+        /// Calculates the checksum of the data
+        /// </summary>
+        /// <returns>byte which represents the checksum of the sentence</returns>
+        public byte CalculateChecksum()
+        {
+            return CalculateChecksumFromSentenceString(ToString());
+        }
+
+        /// <summary>
+        /// Compares sentence identifier with all known identifiers.
+        /// If found returns typed object corresponding to the identifier.
+        /// If not found returns null.
+        /// </summary>
+        /// <returns>Object corresponding to the identifier</returns>
+        public object TryGetTypedValue()
+        {
+            if (s_registeredSentences.TryGetValue(Id, out Func<TalkerSentence, object> producer))
+            {
+                return producer(this);
+            }
+
+            return null;
+        }
+
+        /// <summary>
+        /// Registers sentence identifier as known. Registered sentences are used by <see cref="TryGetTypedValue()"/>.
+        /// </summary>
+        /// <param name="id">NMEA0183 sentence identifier</param>
+        /// <param name="producer">Function which produces typed object given <see cref="TalkerSentence"/>.</param>
+        public static void RegisterSentence(SentenceId id, Func<TalkerSentence, object> producer)
+        {
+            s_registeredSentences[id] = producer;
+        }
+
+        private static (byte? checksum, string lastField) GetChecksumAndLastField(string lastEntry)
+        {
+            int lastStarIdx = lastEntry.LastIndexOf('*');
+
+            if (lastStarIdx == -1 || lastStarIdx != lastEntry.Length - 3)
+            {
+                // there is no checksum, last entry is the last field
+                return (null, lastEntry);
+            }
+
+            int lastIdx = lastEntry.Length - 1;
+            int sum = HexDigitToDecimal(lastEntry[lastIdx - 1]) * 16 + HexDigitToDecimal(lastEntry[lastIdx]);
```

Comment:
NIT: It would be great if you add a small comment here to point to the page in the spec of how the checksum must be calculated in case it ever needs to be changed/fixed.